src/libmongoc/doc/index.rst

```diff
@@ -10,7 +10,7 @@ Introduction
 
 The MongoDB C Driver, also known as "libmongoc", is a library for using MongoDB from C applications, and for writing MongoDB drivers in higher-level languages.
 
-It depends on :doc:`libbson <bson:index>` to generate and parse BSON documents, the native data format of MongoDB.
```

Comment:
This change may resolve the warning, but I expect does not fix the link. The `bson:` targets are for intersphinx in [conf.py](https://github.com/mongodb/mongo-c-driver/blob/59443b78c6336fedaf9d0b850634b11d5b24fc1b/src/libmongoc/doc/conf.py#L78). Regenerating the `libbson.inv` file appears to resolve. Here is a PR to update the libbson.inv file: https://github.com/mongodb/mongo-c-driver/pull/1610